### PR TITLE
Check for naming length restriction before building proposal

### LINKED ISSUE #1654: propose_to_be_provider extrinsic should reject bad names
# Problem 
I was able to add a motion via `propose_to_be_provider` with an invalid provider name (`somethingverylongthatIknowwill fail` to be specific), because it's just a passthrough.  The extrinsic would fail on execution, however, it's undesirable to have people create trash proposals for a number of reasons.

## expected:
the transaction to fail

## actual
the transaction succeeded and a Council motion was created.  

# Solution
Apply existing restrictions to `propose_to_be_provider` which would fail if the same provider name were submitted via `create_provider`

### PR DESCRIPTION
**Pull request by saraswatpuneet**
# Goal
The goal of this PR is to check for provider name length before bundling a proposal that might eventually fail upon execution due to provider name being set is too long 

Closes #1654 

# Checklist
- [ ] Chain spec updated
- [ ] Custom RPC OR Runtime API added/changed? Updated js/api-augment.
- [ ] Design doc(s) updated
- [x] Tests added
- [ ] Benchmarks added
- [ ] Weights updated

### CODE REVIEW

File: pallets/msa/src/lib.rs [hunk 1]
```diff
@@ -886,13 +886,16 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			provider_name: Vec<u8>,
 		) -> DispatchResult {
+			let bounded_name: BoundedVec<u8, T::MaxProviderNameSize> =
+				provider_name.try_into().map_err(|_| Error::<T>::ExceedsMaxProviderNameSize)?;
+
```
Comment (saraswatpuneet) [inline pallets/msa/src/lib.rs:891]: preliminary check for name length, even if actual extrinsic `create_provider` checks for length, a proposal creator wont know until proposal goes though.

File: pallets/msa/src/tests/governance_tests.rs [hunk 2]
```diff
@@ -121,3 +121,17 @@ fn propose_to_be_provider_happy_path() {
...
+	new_test_ext().execute_with(|| {
+		// Create a new MSA account and request that it become a provider
+		let (_new_msa_id, key_pair) = create_account();
+		let proposal_res = Msa::propose_to_be_provider(
+			RuntimeOrigin::signed(key_pair.public().into()),
+			Vec::from("this_is_a_really_long_name_that_should_fail"),
+		);
+
+		assert_noop!(proposal_res, Error::<Test>::ExceedsMaxProviderNameSize);
+	})
```
Comment (saraswatpuneet) [inline pallets/msa/src/tests/governance_tests.rs:136]: Simple test making sure we reject a proposal creation if naming length does not match set limit

**Review by wilwade — APPROVED** [review]
Outside of having to wait for #1961, this looks nice and clean to me.

- Reviewed changes

**Review by mattheworris — APPROVED** [review]
- Read through changes
- Executed e2e tests: 180 passing
- Executed unit tests: all passing
lgtm!

**Review by JoeCap08055 — APPROVED** [review]
🚢 it!

**Pull request merged**